test/scratchOrg.ts

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import child_process from 'child_process';
+import fs from 'fs';
+import path from 'path';
+import util from 'util';
+import {
+  DefaultTreeItem,
+  InputBox,
+  QuickOpenBox,
+  Workbench
+} from 'wdio-vscode-service';
+import {
+  EnvironmentSettings
+} from './environmentSettings';
+import * as utilities from './utilities';
+
+
+const exec = util.promisify(child_process.exec);
+
+export class ScratchOrg {
+  private testSuiteSuffixName: string;
+  private reuseScratchOrg = false;
+  public tempFolderPath: string | undefined = undefined;
+  public projectFolderPath: string | undefined = undefined;
+  private prompt: QuickOpenBox | InputBox | undefined;
+  private scratchOrgAliasName: string | undefined;
+
+  public constructor(testSuiteSuffixName: string, reuseScratchOrg: boolean) {
+    this.testSuiteSuffixName = testSuiteSuffixName;
+    this.reuseScratchOrg = reuseScratchOrg;
+
+    // To have all scratch orgs be reused, uncomment the following line:
+    // this.reuseScratchOrg = true;
+  }
+
+  public get tempProjectName(): string {
+    return 'TempProject-' + this.testSuiteSuffixName;
+  }
+
+  public async setUp(scratchOrgEdition: string = 'Developer'): Promise<void> {
+    await this.setUpTestingEnvironment();
+    await this.createProject(scratchOrgEdition);
+    await this.authorizeDevHub();
+    await this.createDefaultScratchOrg();
+  }
+
+  public async tearDown(): Promise<void> {
+    if (this.scratchOrgAliasName && !this.reuseScratchOrg) {
+      // To use VS Code's Terminal view to delete the scratch org, use:
+      // const workbench = await (await browser.getWorkbench()).wait();
+      // await utilities.executeCommand(workbench, `sfdx force:org:delete -u ${this.scratchOrgAliasName} --noprompt`);
+
+      // The Terminal view can be a bit unreliable, so directly call exec() instead:
+      await exec(`sfdx force:org:delete -u ${this.scratchOrgAliasName} --noprompt`);
+    }
+
+    // This used to work...
+    // if (this.projectFolderPath) {
+    //   await utilities.removeFolder(this.projectFolderPath);
+    // }
+    // ...but something recently changed and now, removing the folder while VS Code has the folder open
+    // causes VS Code to get into a funky state.  The next time a project is created, we get the
+    // following error:
+    //   07:45:19.65 Starting SFDX: Create Project
+    //   ENOENT: no such file or directory, uv_cwd
+    //
+    // Not deleting the folder that was created is OK, b/c it is deleted in setUpTestingEnvironment()
+    // the next time the test suite runs.  I'm going to leave this in for now in case this gets fixed
+    // and this code can be added back in.
```

Comment:
Can this be stashed and saved and reintroduced later if it becomes relevant again?  Do you have any info as to when this could potentially be fixed?

Reply:
Let's chat about this - I normally don't like lots of comments, but (future) dev could review this code and say to themself, "hey, there's a bug here, it should be deleting the files when it cleans up", which, would lead to a bug which this is preventing.  So in this case, I think it makes sense to leave a comment here.